### PR TITLE
Flatten Nft transfer chain

### PR DESCRIPTION
**Pull request by Nana-EC**
Signed-off-by: Nana-EC <nana@swirldslabs.com>

**Description**:
NftTransfer chain for multi recipients can be simplified
- Add merge logic in SqlEntityListener  to flatten

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by xin-hedera — APPROVED** [review]
LGTM

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java [hunk 1]
```diff
@@ -218,6 +221,7 @@ private void cleanup() {
             nfts.clear();
             nftAllowances.clear();
             nftAllowanceState.clear();
+            nftTransferState.clear();
             nftTransfers.clear();
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:225]: nftTransfers is no longer needed

Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java]: Fixed

**Review by xin-hedera — APPROVED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java [hunk 2]
```diff
@@ -560,6 +561,15 @@ private Nft mergeNft(Nft cachedNft, Nft newNft) {
         return cachedNft;
     }
 
+    private NftTransfer mergeNftTransfer(NftTransfer cachedNftTransfer, NftTransfer newNftTransfer) {
+        // flatten multi receiver transfers
+        if (newNftTransfer.getReceiverAccountId() != null && cachedNftTransfer.getReceiverAccountId() != newNftTransfer.getReceiverAccountId()) {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:566]: Is the if even necessary? Do we have scenarios where they won't differ?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java (hunk 2 above)
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:566]: Not that I know of but new `Objects.equals` logic fixes this

[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java:566]: Can adopt `Objects.equals`.
A test handling duplicates would satisfy the test criteria you're referring to. Will add

[the remark this answers is not in the document]
Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java]: Done

Comment (Nana-EC) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java]: Done

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**